models/agent.go

```diff
@@ -211,3 +212,43 @@ func (q *QanAgent) DSN(service *MySQLService) string {
 	// TODO Other parameters?
 	return cfg.FormatDSN()
 }
+
+// AgentFilters represents filters for agents list.
+type AgentFilters struct {
+	ServiceID *string
+}
+
+// AgentsByFilters returns agents providing insights for a given filters.
+func AgentsByFilters(q *reform.Querier, filters AgentFilters) ([]*AgentRow, error) {
+	var agentIDs []interface{}
+	var structs []reform.Struct
+	var err error
+	if filters.ServiceID != nil {
+		agentServices, err := q.SelectAllFrom(AgentServiceView, "WHERE service_id = ?", *filters.ServiceID)
+		if err != nil {
+			return nil, errors.WithStack(err)
+		}
+		for _, str := range agentServices {
+			agentIDs = append(agentIDs, str.(*AgentService).AgentID)
+		}
+
+		if len(agentIDs) == 0 {
+			return []*AgentRow{}, nil
+		}
+
+		structs, err = q.FindAllFrom(AgentRowTable, "id", agentIDs...)
```

Comment:
We need results there to be sorted by ID too. So `SelectAllFrom` with a custom tail.